### PR TITLE
Add more endpoints and tests

### PR DESCRIPTION
**Pull request by namtruong**
Add unit tests for the endpoints, and refactor example packages

### CODE REVIEW

File: nexus-app/src/main/java/com/github/nexus/api/TransactionResource.java [hunk 1]
```diff
@@ -33,6 +44,16 @@ public Response send(@Valid final SendRequest sendRequest){
         return Response.status(Response.Status.CREATED).build();
     }
 
+    @POST
+    @Path("/sendraw")
+    public Response sendRaw(@Context final HttpHeaders headers, InputStream inputStream) throws IOException {
```
Comment (prd-fox) [inline nexus-app/src/main/java/com/github/nexus/api/TransactionResource.java:49]: InputStream should be final

File: nexus-app/src/test/java/com/github/nexus/service/TransactionServiceTest.java [hunk 2]
```diff
@@ -15,4 +15,19 @@ public void testSend(){
     public void testReceive(){
         transactionService.receive();
     }
+
+    @Test
+    public void testDelete(){
+        transactionService.delete();
```
Comment (prd-fox) [inline nexus-app/src/test/java/com/github/nexus/service/TransactionServiceTest.java:21]: Add some verification that the method was actually called?

**Review by melowe — APPROVED** [review]

File: nexus-app/src/test/java/com/github/nexus/service/TransactionServiceTest.java (hunk 2 above)
Comment (namtruong) [reply nexus-app/src/test/java/com/github/nexus/service/TransactionServiceTest.java:21]: the verification should be at the upper level. this test should test what the method does - which is nothing at the mo

Comment (prd-fox) [reply nexus-app/src/test/java/com/github/nexus/service/TransactionServiceTest.java:21]: Oh right yes. Oops.

**Pull request merged**